### PR TITLE
WIP: Test snapshot v1 apis

### PR DESCRIPTION
**Pull request by xing-yang**
**What type of PR is this?**
/kind cleanup

**What this PR does / why we need it**:
This PR updates e2e tests to use snapshot v1 APIs.

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:
```release-note
NONE
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:
```docs

```

### CODE REVIEW

File: test/utils/image/manifest.go [hunk 1]
```diff
@@ -30,6 +30,7 @@ type RegistryList struct {
 	GcAuthenticatedRegistry string `yaml:"gcAuthenticatedRegistry"`
 	DockerLibraryRegistry   string `yaml:"dockerLibraryRegistry"`
 	DockerGluster           string `yaml:"dockerGluster"`
+        DockerTestSnapshot      string `yaml:"dockerTestSnapshot"`
```
Comment (msau42) [inline test/utils/image/manifest.go:33]: I think there's a change needed to add the registry to a case statement:
https://github.com/kubernetes/kubernetes/pull/91257/files#diff-fa4ddc33013d1e57259726a4c1c327903e058453aece262eecfb48300e946c93R292

Comment (xing-yang) [reply test/utils/image/manifest.go]: Done

File: test/e2e/testing-manifests/storage-csi/hostpath/hostpath/csi-hostpath-provisioner.yaml [hunk 2]
```diff
@@ -40,6 +40,7 @@ spec:
       serviceAccountName: csi-provisioner
       containers:
         - name: csi-provisioner
+          #image: docker.io/xyang105/csi-provisioner:apiv1
```
Comment (msau42) [inline test/e2e/testing-manifests/storage-csi/hostpath/hostpath/csi-hostpath-provisioner.yaml:43]: For the provisioner arg error, I think we need to remove https://github.com/kubernetes/kubernetes/blob/master/test/e2e/storage/utils/deployment.go#L92

Comment (xing-yang) [reply test/e2e/testing-manifests/storage-csi/hostpath/hostpath/csi-hostpath-provisioner.yaml:43]: I see.  That makes sense now.

**Comment by xing-yang** [comment]
Closing this PR.

Opened a new one here: https://github.com/kubernetes/kubernetes/pull/96422

**Pull request closed without merging**